packages/xml-views-validation/src/validators/elements/use-of-deprecated-class.ts

```diff
@@ -0,0 +1,54 @@
+import { XMLElement } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { xmlToFQN } from "@ui5-language-assistant/logic-utils";
+import { UseOfDeprecatedClassIssue } from "../../../api";
+
+export function validateUseOfDeprecatedClass(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): UseOfDeprecatedClassIssue[] {
+  const elementTagFqn = xmlToFQN(xmlElement);
+  const ui5Class = model.classes[elementTagFqn];
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  if (ui5Class.deprecatedInfo !== undefined) {
+    const issues: UseOfDeprecatedClassIssue[] = [];
+    const commonIssueFields: UseOfDeprecatedClassIssue = {
+      kind: "UseOfDeprecatedClass",
+      message: `Deprecated UI5 Class: ${elementTagFqn} used.`,
```

Comment:
Did you copy the message from FS? If not I think `Class ${elementTagFqn} is deprecated` sound better

Reply:
Good suggestion

Reply:
The message in the original validations is not precise:
https://github.wdf.sap.corp/DevX/xmlvalidator/blob/master/src/fioriAnalysis/rules/XML_DEPRECATION.js#L79-L83

As the deprecated tag is a UI5 Class not always a control (may also be an element).

I will try to add the additional info to the message as well.

Reply:
Maybe if we change "control" to "class". I like the idea of adding the `since` and `message` to the issue.